packages/react-select/src/__tests__/Select.test.tsx

```diff
@@ -3112,3 +3112,21 @@ test('renders with custom theme', () => {
     window.getComputedStyle(firstOption!).getPropertyValue('background-color')
   ).toEqual(primary);
 });
+
+test('form validation with `required` prop', () => {
+  const components = (value?: Option) => (
+    <form id="formTest">
+      <Select {...BASIC_PROPS} required value={value || null} />
```

Comment:
For testing, it might be better to test that the value is explicitly passed as null instead of coalescing in case the behavior differs between undefined and null. 

Not that it would here, but it's possible some future use-case changes the default behavior and it gets missed.